### PR TITLE
[Feature] confirm exit on close

### PR DESCRIPTION
**Pull request by Wizard603**
Added confirmation message when trying to exit without saving. I needed to update MetricDeleter, MetricAdder, and DataLogger to use the MetricStorage methods because they affect saved status which is needed for the confirmation message to work. Additionally I had to remove the unnecessary Exceptions and added some more comments along the way.

### CODE REVIEW

**Review by craftedswift — DISMISSED** [review]

**Review by Bensonwu02 — APPROVED** [review]
Looks good. thanks for cleaning up the exception handling and correcting the changes to metric summary screen

**Review by craftedswift — DISMISSED** [review]
Removal of Exceptions of AddMetric use case seems to be implemented correctly

**Review by tylerkj1209 — DISMISSED** [review]
Everything looks good. Thanks for removing redundant method preexistingdatapoint, I wasn't sure if we would need that later.

File: src/main/java/entities/MetricStorageInterface.java [hunk 1]
```diff
@@ -71,11 +72,6 @@ public interface MetricStorageInterface {
      */
     File getPath();
 
-    /**
```
Comment (craftedswift) [inline src/main/java/entities/MetricStorageInterface.java:74]: Add this part back in

**Review by craftedswift — APPROVED** [review]

**Review by craftedswift — DISMISSED** [review]

**Review by craftedswift — APPROVED** [review]

**Review by Bensonwu02 — APPROVED** [review]
Thanks for adding the confirm on exit feature. Great extension to our save feature

**Review by tylerkj1209 — APPROVED** [review]
Looks good. I see metri adder formatting was changed but the functionality looks identical.

**Pull request merged**